CRM/Admin/Form/ContactType.php

```diff
@@ -118,27 +119,26 @@ public static function formRule($fields, $files, $self) {
    * Process the form submission.
    */
   public function postProcess() {
-    CRM_Utils_System::flushCache();
-
     if ($this->_action & CRM_Core_Action::DELETE) {
       try {
         CRM_Contact_BAO_ContactType::deleteRecord(['id' => $this->_id]);
+        CRM_Utils_System::flushCache();
         CRM_Core_Session::setStatus(ts('Selected contact type has been deleted.'), ts('Record Deleted'), 'success');
       }
       catch (CRM_Core_Exception $e) {
         CRM_Core_Session::setStatus(ts("Selected contact type can not be deleted. Make sure contact type doesn't have any associated custom data or group."), ts('Sorry'), 'error');
       }
       return;
     }
-    // store the submitted values in an array
+
     $params = $this->exportValues();
 
+    if (!isset($params['is_active'])) {
+      $params['is_active'] = 0;
+    }
```

Comment:
Is this a workaround for using "checkbox" instead of "advcheckbox" on L52?
Switching to "advcheckbox" is probably better & would also probably solve whatever problem CRM-6471 was referring to.

Reply:
Using `advcheckbox` seems to work. I tested enabling/disabling etc.

Is there any reason for not using it systematically? (or alias one to the other?)

Reply:
It's just an ancient thing that was never made consistent.